crates/sui-core/src/authority.rs

```diff
@@ -2423,6 +2423,46 @@ impl AuthorityState {
         }
     }
 
+    pub fn get_checkpoints(&self, reverse: bool) -> Result<Vec<Checkpoint>, anyhow::Error> {
+        let max_checkpoint = self.get_latest_checkpoint_sequence_number()?;
+
+        let values = (0..=max_checkpoint).collect::<Vec<_>>();
+        let mut checkpoint_numbers = values;
+
+        if reverse {
+            checkpoint_numbers.reverse();
+        }
+
+        let verified_checkpoints = self
+            .get_checkpoint_store()
+            .multi_get_checkpoint_by_sequence_number(checkpoint_numbers.as_slice())?;
```

Comment:
If there is a million checkpoints, this will fetch a million checkpoints from the DB. You should add `cursor` and `limit` to this function as well and only `multi_get` `cursor` to `limit + 1`. Please reference the `get_transactions` implementation in this file